ledger/src/blocktree.rs

```diff
@@ -367,8 +360,7 @@ impl Blocktree {
         leader_schedule: Option<&Arc<LeaderScheduleCache>>,
     ) -> Result<()> {
         let db = &*self.db;
-        let mut batch_processor = self.batch_processor.write().unwrap();
```

Comment:
Removing this lock is potentially dangerous if multiple threads (like window_service and broadcast stage) are both trying to write to blocktree at the same time. 

More specifically it opens windows where two threads can interleave read/writes on shared objects like `SlotMetas`